crmsh/ocfs2.py

```diff
@@ -0,0 +1,276 @@
+        self.yes_to_all = context.yes_to_all
+        self.exist_ra_id_list = []
+        self.ra_list = []
+        self.vg_id = None
+        self.target_device = None
+
+    def _verify_packages(self, use_clvm2=False):
+        """
+        Find if missing required package
+        """
+        required_packages = ["ocfs2-tools"]
+        if use_clvm2:
+            required_packages.append("lvm2-lockd")
+        for pkg in required_packages:
+            if not utils.package_is_installed(pkg):
+                raise ValueError("Missing required package for configuring OCFS2: {}".format(pkg))
+
+    def _static_verify(self):
+        """
+        Verify before configuring on init process
+        """
+        self._verify_packages(self.use_clvm2)
+
+        if self.use_stage and self.yes_to_all and not self.ocfs2_devices:
+            raise ValueError("ocfs2 stage require -o option")
+        if len(self.ocfs2_devices) > 1 and not self.use_clvm2:
+            raise ValueError("Without Cluster LVM2 (-C option), -o option only support one device")
+        for dev in self.ocfs2_devices:
+            if not utils.is_block_device(dev):
+                raise ValueError("{} doesn't look like a block device".format(dev))
+            if utils.has_vg_on_disk(dev):
+                raise ValueError("{} belongs to an existing volume group".format(dev))
+            if utils.has_disk_mounted(dev):
+                raise ValueError("{} already mounted".format(dev))
+
+    def _static_verify_on_join(self, dev_list, peer, use_clvm2=False):
+        """
+        Verify before configuring on join process
+        """
+        self._verify_packages(use_clvm2)
+
+        for dev in dev_list:
+            if not utils.is_block_device(dev):
+                raise ValueError("{} doesn't look like a block device".format(dev))
+            local_uuid = utils.get_dev_uuid(dev)
+            peer_uuid = utils.get_dev_uuid(dev, peer)
+            if local_uuid != peer_uuid:
+                raise ValueError("UUID of {} not same with peer {}".format(dev, peer))
+
+    def _dynamic_raise_error(self, error_msg):
+        """
+        Customize error message after cluster running
+        """
+        raise ValueError(error_msg + ("" if self.use_stage else self.HINTS_WHEN_RUNNING))
+
+    def _dynamic_verify(self):
+        """
+        Verify after cluster running
+        """
+        if not utils.has_stonith_running():
+            self._dynamic_raise_error("OCFS2 requires stonith device configured and running")
+
+        if utils.service_is_enabled("sbd.service"):
+            sbd_device_list = bootstrap.SBDManager.get_sbd_device_from_config()
+            for dev in self.ocfs2_devices:
+                if dev in sbd_device_list:
+                    self._dynamic_raise_error("{} cannot be the same with SBD device".format(dev))
+
+        for dev in self.ocfs2_devices:
+            fs_type = utils.get_fs_on_device(dev)
```

Comment:
Do improve the design to detect more than just as a file system. Any other used device is under risk too, eg. partitions, lvm, etc. 

Basically, to change to a different function, ie. utility.is_blk_dev_under_use()